flake.nix

```diff
@@ -0,0 +1,76 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/release-23.11";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    utils,
+    rust-overlay,
+  }:
+    utils.lib.eachDefaultSystem (system: let
+      overlays = [(import rust-overlay)];
+      pkgs = import nixpkgs {inherit system overlays;};
+
+      # Set up the custom tooling based on the rust-toolchain.toml file
+      toolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+      rustPlatform = pkgs.makeRustPlatform {
+        cargo = toolchain;
+        rustc = toolchain;
+      };
+
+      commonDeps = with pkgs; [
+        protobuf
+      ];
+      darwinDeps = with pkgs;
+        lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+          # TODO: Why are the following frameworks required when we aren't hooking into
+          # MacOS internals?
+          Security
+          SystemConfiguration
+        ]);
+    in {
+      # Expose the main router as a buildable package
+      packages.default = rustPlatform.buildRustPackage {
+        pname = "router";
+        version = "1.38.0";
```

Comment:
If we specify the version here then we should also add something into xtask for the release prep because we're doing some batch edits when upgrading the version to be sure it's upgraded too

Reply:
Yeah, I've seen other projects move the version number into its own file and do some shenanigans to include it in Cargo for this reason.